algorithms/random.go

```diff
@@ -0,0 +1,72 @@
+package algorithms
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+var rng = struct {
+	sync.Mutex
+	rand *rand.Rand
+}{
+	rand: rand.New(rand.NewSource(time.Now().UnixNano())),
+}
+
+// Int returns a non-negative pseudo-random int.
+func Int() int {
+	rng.Lock()
+	defer rng.Unlock()
+	return rng.rand.Int()
+}
+
+// Intn generates an integer in range [0,max).
+// By design this should panic if input is invalid, <= 0.
+func Intn(max int) int {
```

Comment:
we should also have that flip function, eg:

```go
func Coinflip(num, dem int) bool {
   r := Intn(dem)
   return r < num
}
```

(obviously it's not hard to do from these, but still nice to have)

Reply:
done